modyn/trainer_server/internal/trainer/remote_downsamplers/remote_gradnorm_downsampling.py

```diff
@@ -49,10 +49,16 @@ def inform_samples(
         target: torch.Tensor,
         embedding: torch.Tensor | None = None,
     ) -> None:
+        if forward_output.dim() == 1:
+            # BCEWithLogitsLoss requires that forward_output and target have the same shape
+            forward_output = forward_output.unsqueeze(1)
+            target = target.unsqueeze(1)
+
         last_layer_gradients = self._compute_last_layer_gradient_wrt_loss_sum(
             self.per_sample_loss_fct, forward_output, target
         )
-        scores = torch.norm(last_layer_gradients, dim=-1).cpu()
+        # pylint: disable=not-callable
+        scores = torch.linalg.vector_norm(last_layer_gradients, dim=1).cpu()
```

Comment:
torch.norm is already deprecated https://pytorch.org/docs/stable/generated/torch.norm.html

Reply:
However, you changed the dim parameter here from -1 to 1. Is that the expected behavior?

Reply:
Yes I validated this before.

`last_layer_gradients` is of two dimensions batch_size * gradient_size. We should calculate the norm only on the dimension 1.

Reply:
dim=-1 refers to the last dimension. So they refer to the same thing (assuming it has 2 dimensions which is true)

Reply:
Makes sense!